File: leetcode/12_sliding_window/1004_Max_Consecutive_Ones_III.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        left, ans, curr = 0, 0, 0
        for right in range(len(nums)):
            if nums[right] == 0:
                curr += 1
            if curr > k:
                if nums[left] == 0:
                    curr -= 1
                left += 1
            ans = max(ans, right - left + 1)
        return ans

File: leetcode/12_sliding_window/1004_Max_Consecutive_Ones_III.go
func longestOnes(nums []int, k int) int {
	var left, ans, cur int
	for right := 0; right < len(nums); right++ {
		if nums[right] == 0 {
			curr++
		}
		if curr > k {
			if nums[left] == 0 {
				curr--
			}
			left++
		}
		ans = max(ans, right-left+1)
	}
	return ans
}
